### PR TITLE
Add monitoring to individual functions in sync

### PR DESCRIPTION
**Pull request by aidenkeating**
This adds time monitoring to many of the functions in fh-mbaas-api.

The monitoring sorted each record by `module` and usually by method
name or by a description in the value for the `method` key of the
record.

I don't know if it makes sense to merge this into the current `monitoring` branch or not. Just adding a PR here for review purposes.

@david-martin @wtrocki Could you take a look please?

### CODE REVIEW

File: lib/sync-DataSetModel.js [hunk 1]
```diff
@@ -350,13 +404,21 @@ var self = {
   },
 
   getClientHash: function (query_params, meta_data) {
+    var startTime = Date.now();
+
     var queryParamsHash = syncUtil.generateHash(query_params);
     var metaDataHash = syncUtil.generateHash(meta_data);
 
+    var timeTaken = Date.now() - startTime;
+    self.apiMetrics.gauge()(self.apiMetrics + '_api_timing',{method:"doSyncLoop"}, timeTaken);
```
Comment (david-martin) [inline lib/sync-DataSetModel.js]: getClientHash?

File: lib/sync-DataSetModel.js [hunk 2]
```diff
@@ -564,10 +629,15 @@ var self = {
 };
 
 var init = function (apiMetrics) {
+
```
Comment (david-martin) [inline lib/sync-DataSetModel.js:640]: could we call defaultDataHandler.setAPIMetrics here instead of changing the signature?

**Comment by wtrocki** [comment]
Awesome change. Please note that this may affect api timing so you cannot relay on api metrics from previous runs. With such amount of different metrics we should probably change way we display those in graphana. We should probably group by time and use max to show what parts of the code are taking most of the time.

**Comment by david-martin** [comment]
@wtrocki Thanks Wojciech.
We're updating grafana dashboard in https://issues.jboss.org/browse/FH-3074 to have a good view of the most important fns and their execution times during load testing

**Comment by david-martin** [comment]
:+1:

**Comment by wtrocki** [comment]
Closing as oudated

**Pull request closed without merging**